modrinth/updater.go

```diff
@@ -0,0 +1,106 @@
+package modrinth
+
+import (
+	"errors"
+
+	"github.com/comp500/packwiz/core"
+	"github.com/mitchellh/mapstructure"
+)
+
+type mrUpdateData struct {
+	ModID            string `mapstructure:"mod-id"`
+	Versions         int    `mapstructure:"versions"`
+	InstalledVersion string `mapstructure:"installed-version"`
+}
+
+func (u mrUpdateData) ToMap() (map[string]interface{}, error) {
+	newMap := make(map[string]interface{})
+	err := mapstructure.Decode(u, &newMap)
+	return newMap, err
+}
+
+type mrUpdater struct{}
+
+func (u mrUpdater) ParseUpdate(updateUnparsed map[string]interface{}) (interface{}, error) {
+	var updateData mrUpdateData
+	err := mapstructure.Decode(updateUnparsed, &updateData)
+	return updateData, err
+}
+
+type cachedStateStore struct {
+	Mod     Mod
+	Version Version
+}
+
+func (u mrUpdater) CheckUpdate(mods []core.Mod, mcVersion string) ([]core.UpdateCheck, error) {
+	results := make([]core.UpdateCheck, len(mods))
+
+	pack, err := core.LoadPack()
+	if err != nil {
+		return results, err
+	}
+
+	for i, mod := range mods {
+		rawData, ok := mod.GetParsedUpdateData("modrinth")
+		if !ok {
+			results[i] = core.UpdateCheck{Error: errors.New("couldn't parse mod data")}
+			continue
+		}
+
+		data := rawData.(mrUpdateData)
+
+		fetchedMod, err := fetchMod(data.ModID)
+		if err != nil {
+			results[i] = core.UpdateCheck{Error: err}
+			continue
+		}
+
+		if len(fetchedMod.Versions) == data.Versions {
+			//the amount of listed versions hasn't changed. There shouldn't be any update
```

Comment:
This is a bad assumption to make - what if the user manually installs an earlier version? The Modrinth API looks really ugly here, I might complain to geo about this - especially as it requires querying *every version* if you want to find the latest for a specific game version?!

Reply:
It is indeed a bad assumption, although I believe it's needed until Modrinth makes some changes. My thoughts were that if a user installs an earlier version, that's probably because there isn't a version for their minecraft. Eg: the pack is 1.16.5 whilst is 1.16.4. In that case the mod can't be updated until a new version is released. 
It's probably best to see what Modrinth does here.